### PR TITLE
allow hiera driven transport/virtual/hash/conffile

### PR DESCRIPTION
**Pull request by cringdahl**
* add more create_resources in init.pp
* remove overt init->hash requirement

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -10,6 +10,12 @@
 #
 # [*configs*]             - (hash)
 #
+# [*hash*]                - (hash)
+#
+# [*transport*]           - (hash)
+#
+# [*virtual*]             - (hash)
+#
```
Comment (ghoneycutt) [inline manifests/init.pp:18]: missing `conffile` entry

File: manifests/init.pp [hunk 2]
```diff
@@ -10,6 +10,12 @@
 #
 # [*configs*]             - (hash)
 #
+# [*hash*]                - (hash)
```
Comment (ghoneycutt) [inline manifests/init.pp:13]: Would be helpful if you document the ones you are adding. This can be something like.

`A hash of postfix::XYZ resources`

File: manifests/hash.pp [hunk 3]
```diff
@@ -50,6 +50,4 @@
     path    => $name,
     mode    => $mode,
   }
-
-  Class['postfix'] -> Postfix::Hash[$title]
```
Comment (ghoneycutt) [inline manifests/hash.pp:54]: This seems unrelated and should probably be its own PR.

**Review by ghoneycutt — CHANGES_REQUESTED** [review]
These will need spec tests added. Here is an example at https://github.com/voxpupuli/puppet-postfix/blob/master/spec/classes/postfix_spec.rb#L468

A simpler set of tests would be for each of these additions, you give a param with data and check that `postfix::XYZ { 'data': }` resource is present.

Here's an example of similar code at https://github.com/sensu/sensu-puppet/blob/master/manifests/resources.pp which is iterating over the hash instead of `create_resources()` and is the new way to do things. You dont need to adopt that here as there is existing code that you are matching. Example spec tests at https://github.com/sensu/sensu-puppet/blob/master/spec/classes/resources_spec.rb

**Comment by ghoneycutt** [comment]
@cringdahl Thanks for the contribution!

File: manifests/hash.pp (hunk 3 above)
Comment (cringdahl) [reply manifests/hash.pp:54]: I needed this to get postfix::hash to loop properly.

**Comment by cringdahl** [comment]
Thanks for the pointers and the New Way example link. Much appreciated.

**Comment by ghoneycutt** [comment]
@cringdahl Awesome work! Could you please rebase and squash your commits into one and I'll merge this.

**Comment by cringdahl** [comment]
> @cringdahl Awesome work! Could you please rebase and squash your commits into one and I'll merge this.

hey, @ghoneycutt , just a quick ping to get this merged in, thanks

**Review by ghoneycutt — APPROVED** [review]
